scripts/registerParachain/extrinsics.js

```diff
@@ -0,0 +1,147 @@
+function findSudoError(events) {
+    let sudoError;
+    events.forEach((record) => {
+        const { event } = record;
+        const types = event.typeDef;
+        if (event.method === 'Sudid') {
+            return event.data.forEach((data, index) => {
+                if (types[index].type == 'DispatchResult' && data.isError) {
+                    sudoError = data.toJSON().err.module;
+                }
+            });
+        }
+    });
+    return sudoError;
+}
+
+function findErrorInMeta(metadata, { index: moduleIndex, error: errorIndex }) {
```

Comment:
were you maybe looking for `api.findError()` ? That takes a module error as bytes and returns detailed error info from the metadata. E.g.:
```typescript
api.findError(dispatchError.asModule.toU8a())
```
Found it when looking into error handling today because of a bug: https://github.com/KILTprotocol/ticket/issues/1472

Reply:
Thanks a lot for the hint. Fixed in 39f50fa!